airflow/decorators/base.py

```diff
@@ -296,7 +297,20 @@ class _TaskDecorator(ExpandableFactory, Generic[FParams, FReturn, OperatorSubcla
     @multiple_outputs.default
     def _infer_multiple_outputs(self):
         try:
-            return_type = typing_extensions.get_type_hints(self.function).get("return", Any)
+            # We only care about the return annotation, not anything about the parameters
+            def fake():
+                ...
+
+            fake.__annotations__ = {"return": self.function.__annotations__["return"]}
```

Comment:
```suggestion
            fake.__annotations__ = {"return": self.function.__annotations__.get("return")}
```
Just in case there isn't a return annotation.